lib/rspec/core/configuration.rb

```diff
@@ -2027,9 +2031,18 @@ def configure_group_with(group, module_list, application_method)
         end
       end
 
-      def on_existing_matching_groups(meta)
-        world.all_example_groups.each do |group|
-          yield group if meta.empty? || MetadataFilter.apply?(:any?, meta, group.metadata)
+      def on_existing_matching_groups(meta, scope=:ignore)
```

Comment:
Hooks have some scope-specific behaviour, but helper modules don't have a scope - the optional argument feels ugly.